spec/javascript/packages/document-capture/components/document-side-acuant-capture-spec.tsx

```diff
@@ -0,0 +1,56 @@
+import { DeviceContext, SelfieCaptureContext } from '@18f/identity-document-capture';
+import DocumentSideAcuantCapture from '@18f/identity-document-capture/components/document-side-acuant-capture';
+import { expect } from 'chai';
+import { render } from '../../../support/document-capture';
+
+describe('DocumentSideAcuantCapture', () => {
+  const DEFAULT_PROPS = {
+    errors: [],
+    registerField: () => undefined,
+    value: '',
+    onChange: () => undefined,
+    onError: () => undefined,
+  };
+
+  context('when selfie is _not_ enabled', () => {
+    it('_does_ display a photo upload button', () => {
+      const { queryAllByText } = render(
+        <DeviceContext.Provider value={{ isMobile: true }}>
+          <SelfieCaptureContext.Provider
+            value={{ isSelfieCaptureEnabled: false, isSelfieDesktopMode: false }}
+          >
+            <DocumentSideAcuantCapture {...DEFAULT_PROPS} side="front" />
+            <DocumentSideAcuantCapture {...DEFAULT_PROPS} side="back" />
+          </SelfieCaptureContext.Provider>
+        </DeviceContext.Provider>,
+      );
+
+      const takeOrUploadPictureText = queryAllByText(
+        'doc_auth.buttons.take_or_upload_picture_html',
+      );
+      expect(takeOrUploadPictureText).to.have.lengthOf(2);
+    });
+  });
+
+  context('when selfie _is_ enabled', () => {
+    it('does _not_ display a photo upload button', () => {
+      const { queryAllByText } = render(
+        <DeviceContext.Provider value={{ isMobile: true }}>
+          <SelfieCaptureContext.Provider
+            value={{ isSelfieCaptureEnabled: true, isSelfieDesktopMode: false }}
+          >
+            <DocumentSideAcuantCapture {...DEFAULT_PROPS} side="front" />
+            <DocumentSideAcuantCapture {...DEFAULT_PROPS} side="back" />
+            <DocumentSideAcuantCapture {...DEFAULT_PROPS} side="selfie" />
+          </SelfieCaptureContext.Provider>
+        </DeviceContext.Provider>,
+      );
+
+      const takePictureText = queryAllByText('doc_auth.buttons.take_picture');
+      expect(takePictureText).to.have.lengthOf(3);
+
+      const notExpectedText = 'doc_auth.buttons.take_or_upload_picture_html';
+      expect(queryAllByText(notExpectedText)).to.be.an('array').that.is.empty;
```

Comment:
👍🏻

Reply:
I thought this was nice too! Really readable syntax and we have it elsewhere in our code. However as I was adding tests in d81b08ce347544dfca96fd4e070f2dc7258a9d12, I found that `that.is.empty` returned `true` even when the length of the array was more than 0. In other words...it doesn't seem to do what I'd expect, [or what the docs imply](https://www.chaijs.com/api/bdd/#method_empty). So I updated it to `to.have.lengthOf(0)`. Less pleasant to read, but more accurate results.

Reply:
>  that.is.empty returned true even when the length of the array was more than 0

WHAT. 😮 That is unexpected.  I would be curious to hear more during a 1:1.

Reply:
I'd wonder if it's the lack of parentheses? I'm not sure I'd expect Chai's default optional parentheses to _not work_, but we do use the [`dirty-chai` plugin](https://www.npmjs.com/package/dirty-chai) with the expectation to prefer explicit parantheses.

Reply:
Oh my gosh you're right! I just tested and if I do `.that.is.empty()` it fails where I'd expect it to. If I do `.that.is.empty` it passes.

That is so confusing since the document from Chai I linked displays it without parens.

Reply:
Yeah, that's a TIL, and very confusing. I would have thought `dirty-chai` just makes the parentheses _work_, not that it requires them 😕